### PR TITLE
Add upgrade process for postgres DB version 15

### PR DESCRIPTION
**Pull request by nadavMiz**
### Explain the changes
1. change the Postgres version from 12 to 15
2. added upgrade code for ODF

### Issues: Fixed #xxx / Gap #xxx
1. https://issues.redhat.com/browse/RHSTOR-4788

### Testing Instructions:
1. on regular image change database version in noobaa CR. should automatically upgrade and migrate the data to the new database
2. on ODF should happen automatically on version upgrade

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: pkg/system/phase2_creating.go [hunk 1]
```diff
@@ -1450,6 +1461,187 @@ func (r *Reconciler) UpgradeMigrateDB() error {
...
+			phase = nbv1.UpgradePhaseNone
+		} else {
+			r.Logger.Infof("UpgradePostgresDB: setting phase to %s", nbv1.UpgradePhasePrepare)
+			phase = nbv1.UpgradePhasePrepare
+			r.NooBaa.Status.BeforeUpgradeDbImage = &oldImage
+		}
+
+	case nbv1.UpgradePhasePrepare:
+
+		r.Logger.Infof("UpgradePostgresDB phase nbv1.UpgradePhasePrepare")
```
Comment (jackyalbo) [inline pkg/system/phase2_creating.go:1561]: why not take this print out and just print phase variable?

**Review by jackyalbo — COMMENTED** [review]
small review for now, once you will update this and especially explain what we expect in what phase we can continue (maybe even together)

**Comment by liranmauda** [comment]
@jackyalbo Can we close this one?

**Comment by jackyalbo** [comment]
Closing as a fixed PR clone was already merged: #1264

**Pull request closed without merging**